app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js

```diff
@@ -92,8 +92,56 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
 				teachingCallFormActionCreators.removePreference(teachingAssignment);
 			};
 
-			$scope.updateAssignmentsOrder = function(sortedTeachingPreferenceIds, termContainer) {
-				teachingCallFormActionCreators.updateAssignmentsOrder(sortedTeachingPreferenceIds, $scope.view.state.scheduleId, termContainer.termCode);
+			$scope.raisePriority = function (preference, preferences, termCode) {
+				var sortedTeachingPreferenceIds = [];
+				var indexToSwap = null;
+
+				// Sort by priority
+				var sortedPreferences = _array_sortByProperty(preferences, "priority");
```

Comment:
_array_sortByProperty makes use of the Array.sort method, it just has additional checks/ common quality of life checks.